pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -113,21 +114,24 @@ func (p *Provider) getStaticParameters(ctx context.Context, instanceType *cloudp
 	if err := instanceType.Requirements.Compatible(scheduling.NewRequirements(scheduling.NewRequirement(v1.LabelArchStable, v1.NodeSelectorOpIn, corev1beta1.ArchitectureArm64))); err == nil {
 		arch = corev1beta1.ArchitectureArm64
 	}
-	// TODO: make conditional on either Azure CNI Overlay or pod subnet
-	vnetLabels, err := p.getVnetInfoLabels(ctx, nodeClass)
-	if err != nil {
-		return nil, err
+
+	if options.FromContext(ctx).NetworkPlugin == networkPluginAzure && options.FromContext(ctx).NetworkPluginMode == networkModeOverlay {
+		// TODO: make conditional on pod subnet
+		vnetLabels, err := p.getVnetInfoLabels(ctx, nodeClass)
+		if err != nil {
+			return nil, err
+		}
+		labels = lo.Assign(labels, vnetLabels)
+	}
+	if options.FromContext(ctx).NetworkPolicy == networkDataplaneCilium {
```

Comment:
Need to clearly distinguish between network-dataplane and network-policy settings (even though both can take "cilium" as one of the allowed values). Please introduce network-dataplane, and use it here.

Reply:
Moved this change out to its own pr https://github.com/Azure/karpenter-provider-azure/pull/318. 

I suspect with network dataplane we will have some discussions about if it should be optional or not. Maybe same with this flag. Some clusters don't need overlay networking to be specified.

This flag was not added as a required flag and was excluded from the required validation for that reason. Cilium dataplane is the same for me.